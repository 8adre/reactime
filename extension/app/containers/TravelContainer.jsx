import React, { Component } from 'react';
import Slider from '../components/Slider'




class TravelContainer extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div>
      <div className="travel-container">TravelContainer</div>
      <Slider 
        className="travel-slider" 
        snapshotLength = {this.props.snapshotsLength} 
        handleChangeSnapshot = {this.props.handleChangeSnapshot}
        snapshotIndex = {this.props.snapshotIndex}
      />
      {`snapshot inex in slider ${this.props.snapshotIndex}`}
      </div>
    )

  }
}
export default TravelContainer;
