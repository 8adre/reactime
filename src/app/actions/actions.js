import * as types from '../constants/actionTypes';

export const toggleMode = mode => ({
  type: types.TOGGLE_MODE,
  payload: mode,
});

export const addNewSnapshots = snapshots => ({
  type: types.NEW_SNAPSHOTS,
  payload: {
    snapshots,
    sliderIndex: snapshots.length - 1,
  },
});

export const initialConnect = (snapshots, mode) => ({
  type: types.INITIAL_CONNECT,
  payload: {
    snapshots,
    mode,
    sliderIndex: 0,
    viewIndex: -1,
  },
});

export const setPort = port => ({
  type: types.SET_PORT,
  payload: port,
});

export const emptySnapshots = () => ({
  type: types.EMPTY,
});

export const changeView = index => ({
  type: types.CHANGE_VIEW,
  payload: index,
});

export const changeSlider = index => ({
  type: types.CHANGE_SLIDER,
  payload: index,
});

export const moveBackward = () => ({
  type: types.MOVE_BACKWARD,
  payload: false,
});

export const moveForward = () => ({
  type: types.MOVE_FORWARD,
  payload: false,
});

export const playForward = () => ({
  type: types.MOVE_FORWARD,
  payload: true,
});

export const pause = () => ({
  type: types.PAUSE,
});

export const startPlaying = intervalId => ({
  type: types.PLAY,
  payload: intervalId,
});

export const importSnapshots = newSnaps => ({
  type: types.IMPORT,
  payload: newSnaps,
});
