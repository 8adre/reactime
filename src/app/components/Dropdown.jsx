import React from 'react';
import Select from 'react-select';
import PropTypes from 'prop-types';

const Dropdown = (props) => {
  const { speeds, setSpeed, selectedSpeed } = props;
  return (
    <Select
      className="react-select-container"
      classNamePrefix="react-select"
      value={selectedSpeed}
      onChange={setSpeed}
      options={speeds}
      menuPlacement="top"
    />
  );
};
Dropdown.propTypes = {
  selectedSpeed: PropTypes.shape({ value: PropTypes.number, label: PropTypes.string }).isRequired,
  speeds: PropTypes.arrayOf(PropTypes.object).isRequired,
  setSpeed: PropTypes.func.isRequired,
};

export default Dropdown;
