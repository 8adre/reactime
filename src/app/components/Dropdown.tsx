import React from 'react';
import Select from 'react-select';

interface DropdownProps {
  selectedSpeed: { value: number; label: string },
  speeds: [];
  setSpeed:  () => void;
}

const Dropdown = (props: DropdownProps) => {
  const { speeds, setSpeed, selectedSpeed } = props;
  return (
    <Select
      className="react-select-container"
      classNamePrefix="react-select"
      value={selectedSpeed}
      onChange={setSpeed}
      options={speeds}
      menuPlacement="top"
    />
  );
};

export default Dropdown;
