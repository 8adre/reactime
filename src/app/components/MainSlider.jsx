import React from 'react';
import Slider from 'rc-slider';
import Tooltip from 'rc-tooltip';
import PropTypes from 'prop-types';

const { Handle } = Slider;

const handle = (props) => {
  const {
    value, dragging, index, ...restProps
  } = props;

  return (
    <Tooltip
      prefixCls="rc-slider-tooltip"
      overlay={value}
      visible={dragging}
      placement="top"
      key={index}
    >
      <Handle value={value} {...restProps} />
    </Tooltip>
  );
};

const MainSlider = ({
  snapshotsLength,
  sliderIndex,
  dispatch,
}) =>
  (
    <Slider
      min={0}
      max={snapshotsLength - 1}
      value={sliderIndex}
      onChange={(index) => {
        const newIndex = index === -1 ? 0 : index;
        dispatch({ type: 'changeSlider', payload: newIndex });
        dispatch({ type: 'pause' });
      }}
      handle={handle}
    />
  );

MainSlider.propTypes = {
  snapshotsLength: PropTypes.number.isRequired,
  sliderIndex: PropTypes.number.isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default MainSlider;
