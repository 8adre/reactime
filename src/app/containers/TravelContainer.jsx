import React, { Component } from 'react';

import Slider from '../components/Slider';

class TravelContainer extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div>
        <div className="travel-container">TravelContainer</div>
        <Slider
          className="travel-slider"
          snapshotLength={this.props.snapshotsLength}
          handleChangeSnapshot={this.props.handleChangeSnapshot}
          handleJumpSnapshot={this.props.handleJumpSnapshot}
          snapshotIndex={this.props.snapshotIndex}
        />
        {`travelContainer snapshotIndex ${this.props.snapshotIndex}`}
      </div>
    );
  }
}
export default TravelContainer;
