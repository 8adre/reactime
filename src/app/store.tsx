import React, { useContext } from 'react';

export const StoreContext = React.createContext();

export const useStoreContext = () => useContext(StoreContext);