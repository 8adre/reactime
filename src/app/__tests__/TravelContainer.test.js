/* eslint-disable react/jsx-filename-extension */
import { shallow, configure } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import TravelContainer from '../containers/TravelContainer';
import MainSlider from '../components/MainSlider';
import Dropdown from '../components/Dropdown';
import { useStoreContext } from '../store';
import { moveBackward, moveForward } from '../actions/actions';

configure({ adapter: new Adapter() });

const state = {
  tabs: {
    87: {
      snapshots: [1, 2, 3, 4],
      sliderIndex: 2,
      playing: true,
    },
  },
  currentTab: 87,
};

const dispatch = jest.fn();
jest.mock('../store');
useStoreContext.mockImplementation(() => [state, dispatch]);

let wrapper;

beforeEach(() => {
  wrapper = shallow(<TravelContainer snapshotsLength={2} />);
  useStoreContext.mockClear();
  dispatch.mockClear();
});

describe('<TravelContainer /> rendering', () => {
  test('should render three buttons', () => {
    expect(wrapper.find('button')).toHaveLength(3);
  });
  test('should render one MainSlider', () => {
    expect(wrapper.find(MainSlider)).toHaveLength(1);
  });
  test('should render one Dropdown', () => {
    expect(wrapper.find(Dropdown)).toHaveLength(1);
  });
});

describe('testing the backward-button', () => {
  test('should dispatch action upon click', () => {
    wrapper.find('.backward-button').simulate('click');
    expect(dispatch.mock.calls.length).toBe(1);
  });

  test('should send moveBackward action to dispatch', () => {
    wrapper.find('.backward-button').simulate('click');
    expect(dispatch.mock.calls[0][0]).toEqual(moveBackward());
  });
});

describe('testing the forward-button', () => {
  test('should dispatch action upon click', () => {
    wrapper.find('.forward-button').simulate('click');
    expect(dispatch.mock.calls.length).toBe(1);
  });

  test('should send moveforward action to dispatch', () => {
    wrapper.find('.forward-button').simulate('click');
    expect(dispatch.mock.calls[0][0]).toEqual(moveForward());
  });
});

describe('testing the play-button', () => {
  test("should display 'pause' if playing is true", () => {
    state.tabs[87].playing = true;
    wrapper = shallow(<TravelContainer snapshotsLength={2} />);
    expect(wrapper.find('.play-button').text()).toBe('Pause');
  });

  test('should display play if playing is false', () => {
    state.tabs[87].playing = false;
    wrapper = shallow(<TravelContainer snapshotsLength={2} />);
    expect(wrapper.find('.play-button').text()).toBe('Play');
  });
});

// describe('testing the playback speed', () => {
//   test('if the playback dropdown states 0.5x the speed should be 0.5x', () => {
//     const wrapper = shallow(<TravelContainer {...props} />);
//     wrapper.find('Dropdown').simulate('change', { value: ['val'] });
//     wrapper.find('select').simulate('change', { value: 'hello' });
//     console.log('val', wrapper.find('Dropdown').simulate('select', { value: ['val'] }));
//     expect(wrapper.find('Dropdown').text()).toBe('0.5x');
//     expect(wrapper.find('select [selected]').val()).toEqual('key');
//   });
//   test('if the playback dropdown states 1x the speed should be 1x', () => {
//     const wrapper = shallow(<TravelContainer {...dropdownProps} />);

//     expect(wrapper.find('Dropdown').label).toBe('1.0x');
//   });
//   test('if the playback dropdown states 2x the speed should be 2x', () => {
//     const wrapper = shallow(<TravelContainer {...dropdownProps} />);
//     wrapper.find('Dropdown').simulate('click');
//     expect(wrapper.find('Dropdown').label).toBe('2.0x');
//   });
// });
