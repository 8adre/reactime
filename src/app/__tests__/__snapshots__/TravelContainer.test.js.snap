// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `undefined`;

exports[`<TravelContainer /> rendering should render three buttons 1`] = `ShallowWrapper {}`;
